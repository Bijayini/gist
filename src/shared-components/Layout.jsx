/* eslint-disable import/no-named-as-default */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Header from './Header';

const Layout = ({ children }) => (
  <Fragment>
    <Header />
    <div className="main-layout">{children}</div>
    <style jsx="true" global="true">
      {`
        @font-face {
          font-family: Lato, sans-serif;
          font-display: swap;
        }
        html {
          height: 100%;
          font-size: 16px;
        }
        body {
          margin: 0;
        }

        dl,
        dt,
        dd,
        ul,
        ol {
          list-style-type: none;
          padding: 0;
          margin: 0;
        }
        p {
          margin: 0;
          padding: 0;
        }
        a {
          background-color: transparent;
          text-decoration: none;
        }
        button,
        input {
          font-family: inherit;
          font-size: 100%;
          line-height: 1.15;
          margin: 0;
        }
        html,
        body {
          -webkit-box-sizing: border-box;
          -moz-box-sizing: border-box;
          box-sizing: border-box;
          -webkit-font-smoothing: antialiased;
          -moz-osx-font-smoothing: grayscale;
        }
        input[type='text']::-ms-clear {
          display: none;
        }
        * {
          box-sizing: border-box;
        }
      `}
    </style>
  </Fragment>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Layout;
