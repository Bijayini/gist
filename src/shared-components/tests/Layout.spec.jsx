import React from 'react';
import { shallow } from 'enzyme';

import Layout from '../Layout';

describe('<Layout />', () => {
  let component;

  beforeEach(() => {
    component = shallow(
      <Layout>
        <span>Some data</span>
      </Layout>,
    );
  });

  it('should match with snapshot', () => {
    expect(component.debug()).toMatchSnapshot();
  });
});
