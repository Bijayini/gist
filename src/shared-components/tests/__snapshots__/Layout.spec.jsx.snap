// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should match with snapshot 1`] = `
"<Fragment>
  <Header />
  <div className=\\"jsx-893738437 main-layout\\">
    <span>
      Some data
    </span>
  </div>
  <JSXStyle id=\\"893738437\\">
    @font-face{font-family:Lato,sans-serif;font-display:swap;}html{height:100%;font-size:16px;}body{margin:0;}dl,dt,dd,ul,ol{list-style-type:none;padding:0;margin:0;}p{margin:0;padding:0;}a{background-color:transparent;-webkit-text-decoration:none;text-decoration:none;}button,input{font-family:inherit;font-size:100%;line-height:1.15;margin:0;}html,body{-webkit-box-sizing:border-box;-moz-box-sizing:border-box;box-sizing:border-box;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale;}input[type=&#39;text&#39;]::-ms-clear{display:none;}*{box-sizing:border-box;}
  </JSXStyle>
</Fragment>"
`;
